access/jira/app.go

```diff
@@ -137,60 +112,123 @@ func (a *App) run(ctx context.Context) (err error) {
 		return
 	}
 
-	a.mainJob.SetReady(httpOk && watcherOk)
+	ok := httpOk && watcherOk
+	a.mainJob.SetReady(ok)
+	if ok {
+		log.Info("Plugin is ready")
+	} else {
+		log.Error("Plugin is not ready")
+	}
 
 	<-httpJob.Done()
 	<-watcherJob.Done()
 
 	return trace.NewAggregate(httpJob.Err(), watcherJob.Err())
 }
 
-func (a *App) checkTeleportVersion(ctx context.Context) error {
+func (a *App) init(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, initTimeout)
+	defer cancel()
+	log := logger.Get(ctx)
+
+	var (
+		err  error
+		pong proto.PingResponse
+	)
+
+	bk := backoff.DefaultConfig
+	bk.MaxDelay = backoffMaxDelay
+	if a.apiClient, err = client.New(ctx, client.Config{
+		Addrs:       []string{a.conf.Teleport.AuthServer},
+		Credentials: a.conf.Teleport.Credentials(),
+		DialOpts:    []grpc.DialOption{grpc.WithConnectParams(grpc.ConnectParams{Backoff: bk, MinConnectTimeout: initTimeout})},
+	}); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if pong, err = a.checkTeleportVersion(ctx); err != nil {
+		return trace.Wrap(err)
+	}
+
+	a.jira, err = NewJiraClient(a.conf.JIRA, pong.ClusterName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	log.Debug("Starting JIRA API health check...")
+	if err = a.jira.HealthCheck(ctx); err != nil {
+		return trace.Wrap(err, "api health check failed")
+	}
+	log.Debug("JIRA API health check finished ok")
+
+	// Create webhook server providing a.OnJIRAWebhook as a callback function
+	if a.webhookSrv, err = NewWebhookServer(a.conf.HTTP, a.onJIRAWebhook); err != nil {
+		return trace.Wrap(err)
+	}
+	if err = a.webhookSrv.EnsureCert(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return nil
+}
+
+func (a *App) checkTeleportVersion(ctx context.Context) (proto.PingResponse, error) {
 	log := logger.Get(ctx)
 	log.Debug("Checking Teleport server version")
-	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
-	defer cancel()
-	pong, err := a.accessClient.Ping(ctx)
+	pong, err := a.apiClient.WithCallOptions(grpc.WaitForReady(true)).Ping(ctx)
```

Comment:
This shouldn't be needed with the new client
```suggestion
	pong, err := a.apiClient.Ping(ctx)
```